internal/service/indexer/main.go

```diff
@@ -0,0 +1,81 @@
+package indexer
+
+import (
+	"context"
+
+	"github.com/dov-id/cert-integrator-svc/internal/config"
+	"github.com/dov-id/cert-integrator-svc/internal/data"
+	"github.com/dov-id/cert-integrator-svc/internal/data/postgres"
+	"github.com/dov-id/cert-integrator-svc/internal/helpers"
+	"gitlab.com/distributed_lab/logan/v3"
+	"gitlab.com/distributed_lab/logan/v3/errors"
+)
+
+type IIndexer interface {
```

Comment:
In golang we don't name interfaces with additional I (I mean from interface word), can you make Indexer structure private? Otherwise - rename IIndexer.